src/lib/permissions.ts

```diff
@@ -0,0 +1,79 @@
+import { type StudyStatus } from '@/database/types'
+import { type UserSession } from './types'
+import { AbilityBuilder, MongoAbility, createMongoAbility, subject } from '@casl/ability'
+
+export { subject }
+
+type Record = { id: string }
+type RecordWithOrgId = { orgId: string }
+type Study = { study: { orgId: string } }
+
+type Abilities =
+    | ['invite', 'User']
+    | ['claim', 'PendingUser']
+    | ['update', 'User' | Record]
+    | ['read', 'User' | Record]
+    | ['read' | 'update', 'ReviewerKey' | { userId: string }]
+    | ['read' | 'update' | 'create' | 'delete', 'Team' | RecordWithOrgId]
+    | ['view' | 'update' | 'delete' | 'review', 'Study' | Study]
+    | ['read' | 'create', 'StudyJob' | Study]
+    | ['create', 'Study' | { orgId: string }]
+    | ['approve' | 'reject', 'Study' | { orgId: string; status: StudyStatus }]
+
+export type AppAbility = MongoAbility<Abilities>
+
+export function defineAbilityFor(session: UserSession) {
+    const { isSiAdmin } = session.user
+    const { isAdmin: isTeamAdmin, isResearcher, isReviewer } = session.team
+    const { can: permit, build } = new AbilityBuilder<AppAbility>(createMongoAbility)
+
+    // https://casl.js.org/v6/en/guide/conditions-in-depth
+    // rules use mongodb query conditions: https://casl.js.org/v6/en/guide/conditions-in-depth
+
+    // action all users can perform
+    permit('update', 'User', { id: session.user.id })
+    permit('claim', 'PendingUser')
+
+    permit('read', 'Team', { orgSlug: session.team.slug })
+    permit('read', 'Team', { id: session.team.id })
+
+    if (isResearcher || isReviewer || isTeamAdmin) {
+        permit('view', 'Study', { orgId: session.team.id })
```

Comment:
the use of orgId is different than here:

https://github.com/safeinsights/management-app/blob/facb5277633c987b8c3a232583ba542fec9d78a9/src/lib/permissions.ts#L57

top level `orgID` usage vs `study.orgId` usage. Can this lead to errors?

It seems e.g.:
`getStudyAction` (used by researcher/.../review/page.tsx), the subject passed to the ability check is shaped like `{ study: { orgId: '...' } }` and this check will fail here probably?